velox/vector/tests/utils/VectorTestBase.cpp

```diff
@@ -116,4 +116,24 @@ void assertCopyableVector(const VectorPtr& vector) {
   copy->copy(vector.get(), 0, 0, vector->size());
 }
 
+void assertNoOverlappingRanges(const ArrayVectorPtr& arrayVector) {
```

Comment:
Should we move this into ArrayVector and MapVector's constructor (perhaps, as debug-only check)?